src/jarabe/journal/journalactivity.py

```diff
@@ -241,12 +255,45 @@ def _session_manager_shutdown_cb(self, event):
     def can_close(self):
         return False
 
-    def _setup_main_view(self):
-        self._main_toolbox = MainToolbox()
-        self._edit_toolbox = EditToolbox(self)
-        self._main_view = Gtk.VBox()
-        self._main_view.set_can_focus(True)
-
+    def _create_add_new_entry(self):
+        self._add_new_box = Gtk.Box(orientation = Gtk.Orientation.HORIZONTAL)
+        add_new_button = EventIcon(icon_name='list-add') # suggest icon for this
+        add_new_button.set_tooltip(_('Add New'))
+        add_new_button.connect('button-press-event', self._add_new_button_clicked_cb)
+        self._add_new_box.pack_start(add_new_button, False, True, 0)
+        add_new_button.show()
+
+        self._entry = iconentry.IconEntry()
+        self._entry.connect('key-press-event',self._key_press_add_new_entry_cb)
+        text = _('Add new entry')
+        self._entry.set_placeholder_text(text)
+        self._entry.add_clear_button()
+        self._add_new_box.pack_start(self._entry, True, True, 0)
+        self._entry.show()
+        return self._add_new_box
+
+    def get_add_new_box(self):
+        return self._add_new_box
+
+    def _create_add_new_entry_project(self):
+        hbox = Gtk.Box(orientation = Gtk.Orientation.HORIZONTAL)
```

Comment:
the code is looking good from a macro level, just a few flake8 issues.  running flake8 prints them all out.